src/Stratis.Bitcoin.Features.PoA/PoAFeature.cs

```diff
@@ -62,6 +63,7 @@ public static IFullNodeBuilder UsePoAConsensus(this IFullNodeBuilder fullNodeBui
                         services.AddSingleton<FederationManager>();
                         services.AddSingleton<PoABlockHeaderValidator>();
                         services.AddSingleton<PoAMiner>();
+                        services.AddSingleton<PoABlockDefinition, PoABlockDefinition>();
```

Comment:
Should be `services.AddSingleton<BlockDefinition, PoABlockDefinition>();`

Reply:
BlockDefinition has an interface I don't want to use really. It makes me provide a script. But script is not needed without a subsidy for a block.

Reply:
but you're inheriting from BlockDefinition already.

Reply:
Yeah, but when I inherit I set default value to the script to be null. And in `BlockDefinition` there is no default value so caller will have to guess if putting null there is ok, when in fact that's the only thing that caller should do